azerothcore/azerothcore-wotlk
# fix: Remove hardcoded texts (WIP)

## Changes Proposed:
Remove a lot of hardcoded texts in the C++ sources, and use texts from the database instead (gossips, emotes, etc)
See https://github.com/azerothcore/azerothcore-wotlk/projects/15

## Issues Addressed:

- Closes #4192 
- Closes #4191 
- Closes #4190 
- Closes #4180 
- Closes #4176
- Closes #4169
- Closes #4166
- Closes #4164


## How to Test the Changes:


- Alterac Valley (done ✅):
https://github.com/azerothcore/azerothcore-wotlk/issues/4192
Verify that the NPCs (Balinda Stonehearth, Galvangar, Snivvle) yell:
  * "Begone. Uncouth scum! The Alliance shall prevail in Alterac Valley!"
  * "Now is the time to attack! For the Horde!"
  * "Snivvle is here! Snivvle claims the Coldtooth Mine!"

- Undercity, Parqual Fintallas (done ✅):
https://github.com/azerothcore/azerothcore-wotlk/issues/4191
  * `.go c 32022`
  * `.q add 6628`
  * Talk to Parqual Fintallas and verify that the gossip options are like the following: https://youtu.be/UOwwR1yvroQ?t=306

- Western Plaguelands (done ✅):
https://github.com/azerothcore/azerothcore-wotlk/issues/4190
  * `.go c 45238`
  * `.q add 5226`
  * Turn in [Return to Chillwind Camp]
  * Commander Ashlam Valorfist gives a quest, turn it in
  * Talk to Alchemist Arbington
  * Verify that the gossip options are:
     "What's needed for the cauldron at Felstone Field? / Dalson's Tears? / the Writhing Haunt? / Gahrron's Withering?"
     "I need a vitreous focuser."

- Karazhan **(WIP 🔧)**:
https://github.com/azerothcore/azerothcore-wotlk/issues/4188
  * `.go c 135823`

- Zul'Aman **(WIP 🔧)**:
https://github.com/azerothcore/azerothcore-wotlk/issues/4187
  * Halazzi: `.go c 86195`
  * Akil'zon: `.go c 86494`
  * Nalorakk: `.go c 86609`
  * Jan'alai: `.go c 89322`

  * Harkor: [https://youtu.be/ni2MtU7SHvc?t=567](https://youtu.be/ni2MtU7SHvc?t=567)
Well then, I spose I be owin' ya my thanks
We've killed your captors. You're free to go.

  * Tanzar: [https://youtu.be/ni2MtU7SHvc?t=1194](https://youtu.be/ni2MtU7SHvc?t=1194)
You kill the trollbear, mon?
Nalorakk is dead, you're free to go.

  * Kraz: [https://youtu.be/ni2MtU7SHvc?t=1896](https://youtu.be/ni2MtU7SHvc?t=1896)
Freedom! Kraz life was over
We've killed your captors. You're free to go.

  * Ashli: [https://youtu.be/ni2MtU7SHvc?t=2720](https://youtu.be/ni2MtU7SHvc?t=2720)
Oh hooray - you killed the bad troll just in time to save me
It's safe, little gnome. You can come out now.

- Moonglade (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4180
  * "I'd like to fly to Rut'theran Village." `.go c 42338`
  * "I'd like to fly to Thunder Bluff." `.go c 42337`

  * "Do you know where I can find Half Pendant of Aquatic Endurance?" / "Do you know where I can find Half Pendant of Aquatic Agility?"
     Alliance: `.q add 272` 
     Horde: `.q add 30`
     then `.go c 42337` `.go c 42338`

  * "What do you represent, spirit?"
    "I seek to understand the importance of strength of the body."
    "I seek to understand the importance of strength of the heart."
    "I have heard your words, Great Bear Spirit, and I understand. I now seek your blessings to fully learn the way of the Claw."
       * Alliance: `.q add 5929`
       * Horde: `.q add 5930`
       * then `.go c 42356`

- Tanaris (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4176
  First part:
    * `.go c steward_of_time`
    * Take the quest [To The Master's Lair]
    * `.go c 23284` if you don't want to enjoy the ride
    * Turn in [To The Master's Lair]
    * `.go c steward_of_time`
    * Verify that the gossip option is "Please take me to the master's lair."

  Second part:
    * `.q add 2946`
    * `.go -9619 -2815 15 1`
    * Turn in [Seeing What Happens]
    * Take [The Stone Watcher]
    * Verify the gossip options (https://youtu.be/cT4Zv791uHg?t=409):
      * "What function do you serve?"
      * "What are the Plates of Uldum?"
      * "Where are the Plates of Uldum?"
      * "Excuse me? We've been "reschedueled for visitations"? What does that mean?!"
      * "So, what's inside Uldum?"
      * "I will return when i have the Plates of Uldum."

- The Culling of Stratholme (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4169
    * `.tele cullingofstratholme`
    * Find Arthas (`.go c 1970935`)
    * Progress through the instance, and verify that Arthas' gossip options are like the following:
    * "Yes, my Prince. We are ready."
    * "We're only doing what is best for Lordaeron, your Highness."
    * "Lead the way, Prince Arthas."
    * "I'm ready."
    * For Lordaeron!"
    * "I'm ready to battle the dreadlord, sire."

- Howling Fjord (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4166
  `.q add 11221`
  * "High Executor Anselm wants a report on the situation."
    * `.go c 113843`
  * "High Executor Anselm requests your report."
    * `.go c 117636`
    * `.respawn darkrangerlyana` might help if she died fighting

- Wintergrasp (done ✅)
https://github.com/azerothcore/azerothcore-wotlk/issues/4164
  * `.tele wintergrasp`
  * `.bf start 1`
  * Join the battle
  * `.aura 55629`
  * `.go 4937 2388 330`
  * Verify that you have the following gossip items:
    * "I'd like to build a catapult."
    * "I'd like to build a demolisher."
    * "I'd like to build a siege engine."


## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
